### PR TITLE
MULE-10362: provide doc support for smart connectors

### PR DESCRIPTION
**Pull request by fernandezlautaro**
Connectors in XML now expose the category, vendor and documentation.

### CODE REVIEW

File: modules/spring-config/src/main/resources/META-INF/mule-module.xsd [hunk 1]
```diff
@@ -87,6 +67,58 @@
         </xsd:complexContent>
     </xsd:complexType>
 
+    <xsd:element name="parameter" type="parameterType">
+        <xsd:annotation>
+            <xsd:documentation>
+                An named element that will be consumed only in the scope of the body element by the list of message processors.
```
Comment (pablolagreca) [inline modules/spring-config/src/main/resources/META-INF/mule-module.xsd]: Doesn't seem to accurate this description.

File: modules/spring-config/src/main/resources/META-INF/mule-module.xsd [hunk 2]
```diff
@@ -133,7 +154,35 @@
...
+                    <xsd:annotation>
+                        <xsd:documentation>
+                            Expected category of the module.
+                        </xsd:documentation>
+                    </xsd:annotation>
+                </xsd:attribute>
+                <xsd:attribute name="vendor" type="xsd:string" default="MuleSoft">
+                    <xsd:annotation>
+                        <xsd:documentation>
+                            Expected category of the module.
```
Comment (pablolagreca) [inline modules/spring-config/src/main/resources/META-INF/mule-module.xsd]: wrong doc

File: modules/spring-config/src/main/resources/META-INF/mule-module.xsd (hunk 1 above)
Comment (fernandezlautaro) [inline modules/spring-config/src/main/resources/META-INF/mule-module.xsd]: why not?
a `<parameter>` is the only visible thing any message processor within the `<body>` element can consume as they don't see the event going through

Mind recommending me a different doc for it?

File: modules/spring-config/src/main/resources/META-INF/mule-module.xsd (hunk 2 above)
Comment (fernandezlautaro) [inline modules/spring-config/src/main/resources/META-INF/mule-module.xsd]: done

File: modules/spring-config/src/main/resources/META-INF/mule-module.xsd (hunk 1 above)
Comment (pablolagreca) [inline modules/spring-config/src/main/resources/META-INF/mule-module.xsd]: A parameter element defines an input value for the operation in which it is define. Such parameter must be defined with a meaningful name, a type which defines the kind of content the parameter must have and optionally a default value that will be used if the invocation to the operation does not defines a value for the parameter. The parameter can be accesed within the body definition of the operation using #[..lalalaa..].

**Pull request merged**